src/DynamoCoreWpf/Views/Core/NoteView.xaml.cs

```diff
@@ -152,5 +168,195 @@ private void PrepareZIndex()
                 child.ViewModel.ZIndex = Configurations.NodeStartZIndex;
             }
         }
-   }
+
+        private void noteTextBox_LostFocus(object sender, RoutedEventArgs e)
+        {
+            Panel.SetZIndex(noteTextBox, 0);
+        }
+
+        private void noteTextBox_PreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            Key[] specialKeys = { Key.OemMinus, Key.Tab, Key.Enter };
+            if (!specialKeys.Contains(e.Key))
+                return;
+
+            e.Handled = true;
+            var textBox = sender as TextBox;
+            var text = textBox.Text;
+            var caretIndex = textBox.CaretIndex;
+
+            if (e.Key == Key.OemMinus)
+                textBox.Text = BulletDashHandler(text, caretIndex);
+            
+            if (e.Key == Key.Tab)
+                textBox.Text = BulletTabHandler(text, caretIndex);
+
+            if (e.Key == Key.Enter)
+                textBox.Text = BulletEnterHandler(text, caretIndex);
+
+            textBox.CaretIndex = caretIndex + (textBox.Text.Length - text.Length);
+        }
+
+        #region Bullet point support
+
+        /// <summary>
+        /// List of bullets characters to use in sequence of indentation
+        /// </summary>
+        private static readonly char[] BULLETS_CHARS = { '\u2022', '\u25E6', '\u2023' };
+
+        /// <summary>
+        /// Handles text when user types Key.OemMinus (DASH)
```

Comment:
Maybe add what it actually does, like it converts the dash into a bullet and indent the text

Reply:
@martinstacey could we add a bit more detail to what this does